include/arbor/simd/simd.hpp

```diff
@@ -220,8 +220,15 @@ namespace simd_detail {
                     scalar_type a[width];
                     Impl::copy_to(s, a);
 
+                    scalar_type temp = 0;
                     for (unsigned i = 0; i<width; ++i) {
-                            p[o[i]] += a[i];
+                        unsigned curr_index_ = o[i];
+                        unsigned next_index_ = i == width - 1 ? o[i]+1 : o[i+1];
+                        temp += a[i];
+                        if(curr_index_ != next_index_) {
+                            p[curr_index_] += temp;
+                            temp = 0;
+                        }
                     }
```

Comment:
Fix is sound! I think simplifying (maybe?) the code slightly will save us a comparison and make it a little easier to read, e.g.
```
    scalar_type temp = 0;
    for (unsigned i = 0; i<width-1; ++i) {
        temp += a[i];
        if (o[i] != o[i+1]) {
            p[o[i]] += temp;
            temp = 0;
        }
    }
    p[o[width-1]] = temp;
```

Reply:
That does look better, I modified it :)